Automattic/woocommerce-payments
# refactor disputed order notice to separate side effects & add a `_view` tracks event

Fixes #7437


- ~~alternative to #7545~~ now closed
- see comment on issue for rationale https://github.com/Automattic/woocommerce-payments/issues/7437#issuecomment-1780237772

Please also review the registration PR: Automattic/tracks-events-registration#1921 (private repo)

#### Changes proposed in this Pull Request

- Refactor disputed order notice code to simplify logic. 
- Move notices to a dedicated module, away from the old school jQuery code (simplify the main file).
- Use explicit side effects (`useEffect`) for:
  - manipulating DOM to disable refund button
  - firing a tracks event for notice render `wcpay_order_dispute_notice_view`
- Removes the info about why refund is disabled from the notice. This clutters the notice and may be distracting / confusing for merchants. The purpose of the notice is to drive merchant to respond to dispute.

#### Testing instructions
Test viewing order page with all dispute scenarios: 

- no dispute
- dispute with various days remaining
- inquiry with various days remaining
- dispute with various status (pending, lost, won)
 
Notice should work as before (#6354) and refund behaviour should be disabled appropriately when order is disputed (#7043).

When the order is locked because of a pending dispute, the merchant should have a clue about why – tooltip at bottom of order table should be relevant. The dispute notice should focus on the dispute and the fact that merchant should take action.

When the order has a resolved or pending dispute, there should be a notice explaining that the order is locked and that this is due to the dispute.

-------------------


**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_ na - behaviour is unchanged
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. na - no change to flows
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. na
